app/javascript/components/Profile/Layout.tsx

```diff
@@ -13,55 +13,64 @@ import { PoweredByFooter } from "$app/components/PoweredByFooter";
-            >
-              Impersonate
-            </NavigationButton>
-          ) : null}
-          <img className="user-avatar" src={creatorProfile.avatar_url} alt="Profile Picture" />
-          <a href={Routes.root_path()} style={{ textDecoration: "none" }}>
-            {creatorProfile.name}
-          </a>
-        </section>
-        {!hideFollowForm ? (
-          <section>
-            <FollowForm creatorProfile={creatorProfile} />
-          </section>
-        ) : null}
-        {creatorProfile.twitter_handle || cartItemsCount ? (
-          <section className="links">
-            {creatorProfile.twitter_handle ? (
-              <NavigationButton outline href={`https://twitter.com/${creatorProfile.twitter_handle}`} target="_blank">
-                <Icon name="twitter" />
+    <div className="flex min-h-full flex-col">
+      <header className="relative z-20 border-border bg-background text-lg lg:border-b">
+        <div className="mx-auto flex max-w-6xl flex-wrap lg:flex-nowrap lg:items-center lg:gap-6 lg:py-6">
+          <Section className="relative order-1 grow lg:order-none lg:flex-1">
+            {(loggedInUser?.isGumroadAdmin || loggedInUser?.isImpersonating) &&
+            creatorProfile.external_id !== loggedInUser.id ? (
+              <NavigationButton
+                href={Routes.admin_impersonate_url({ user_identifier: creatorProfile.external_id })}
+                className="absolute left-3"
+                color="filled"
+              >
+                Impersonate
               </NavigationButton>
             ) : null}
-            <CartNavigationButton />
-          </section>
-        ) : null}
+            <img className="user-avatar" src={creatorProfile.avatar_url} alt="Profile Picture" />
+            <a href={Routes.root_path()} className="no-underline">
+              {creatorProfile.name}
+            </a>
+          </Section>
+          {!hideFollowForm ? (
+            <Section className="order-3 basis-full lg:order-none lg:basis-auto">
+              <FollowForm creatorProfile={creatorProfile} />
+            </Section>
+          ) : null}
+          {creatorProfile.twitter_handle || cartItemsCount ? (
+            <Section className="order-2 ml-auto lg:order-none lg:ml-0">
+              {creatorProfile.twitter_handle ? (
+                <NavigationButton outline href={`https://twitter.com/${creatorProfile.twitter_handle}`} target="_blank">
+                  <Icon name="twitter" />
+                </NavigationButton>
+              ) : null}
+              <CartNavigationButton />
+            </Section>
+          ) : null}
+        </div>
       </header>
-      <main>
+      <main className="flex-1">
         {children}
-        <PoweredByFooter />
+        <PoweredByFooter className="mx-auto w-full max-w-6xl lg:py-6 lg:text-left" />
       </main>
     </div>
   );
 };
+
+const Section = ({ children, className }: { children: React.ReactNode; className?: string }) => (
```

Comment:
Instead of this mini-component, could we just make a `const sectionClassNames` string?